src/d3d9/d3d9_interfaces.h

```diff
@@ -0,0 +1,201 @@
+          VkImageCreateInfo*    pInfo) = 0;
+};
+
+/**
+ * \brief D3D9 device interface for Vulkan interop
+ *
+ * Provides access to the device and instance handles
+ * as well as the queue that is used for rendering.
+ */
+MIDL_INTERFACE("2eaa4b89-0107-4bdb-87f7-0f541c493ce0")
+ID3D9VkInteropDevice : public IUnknown {
+  /**
+   * \brief Queries Vulkan handles used by DXVK
+   * 
+   * \param [out] pInstance The Vulkan instance
+   * \param [out] pPhysDev The physical device
+   * \param [out] pDevide The device handle
+   */
+  virtual void STDMETHODCALLTYPE GetVulkanHandles(
+          VkInstance*           pInstance,
+          VkPhysicalDevice*     pPhysDev,
+          VkDevice*             pDevice) = 0;
+
+  /**
+   * \brief Queries the rendering queue used by DXVK
+   * 
+   * \param [out] pQueue The Vulkan queue handle
+   * \param [out] pQueueIndex Queue index
+   * \param [out] pQueueFamilyIndex Queue family index
+   */
+  virtual void STDMETHODCALLTYPE GetSubmissionQueue(
+          VkQueue*              pQueue,
+          uint32_t*             pQueueIndex,
+          uint32_t*             pQueueFamilyIndex) = 0;
+
+  /**
+   * \brief Transitions a Texture to a given layout
+   * 
+   * Executes an explicit image layout transition on the
+   * D3D device. Note that the image subresources \e must
+   * be transitioned back to its original layout before
+   * using it again from D3D9.
+   * \param [in] pTexture The image to transform
+   * \param [in] pSubresources Subresources to transform
+   * \param [in] OldLayout Current image layout
+   * \param [in] NewLayout Desired image layout
+   */
+  virtual void STDMETHODCALLTYPE TransitionTextureLayout(
+          ID3D9VkInteropTexture*    pTexture,
+    const VkImageSubresourceRange*  pSubresources,
+          VkImageLayout             OldLayout,
+          VkImageLayout             NewLayout) = 0;
+
+  /**
+   * \brief Flushes outstanding D3D rendering commands
+   * 
+   * Must be called before submitting Vulkan commands
+   * to the rendering queue if those commands use the
+   * backing resource of a D3D9 object.
+   */
+  virtual void STDMETHODCALLTYPE FlushRenderingCommands() = 0;
+  
+  /**
+   * \brief Locks submission queue
+   * 
+   * Should be called immediately before submitting
+   * Vulkan commands to the rendering queue in order
+   * to prevent DXVK from using the queue.
+   * 
+   * While the submission queue is locked, no D3D9
+   * methods must be called from the locking thread,
+   * or otherwise a deadlock might occur.
+   */
+  virtual void STDMETHODCALLTYPE LockSubmissionQueue() = 0;
+  
+  /**
+   * \brief Releases submission queue
+   * 
+   * Should be called immediately after submitting
+   * Vulkan commands to the rendering queue in order
+   * to allow DXVK to submit new commands.
+   */
+  virtual void STDMETHODCALLTYPE ReleaseSubmissionQueue() = 0;
+
+  /**
+   * \brief Locks the device
+   * 
+   * Can be called to ensure no D3D9 device methods
```

Comment:
You should probably mention that this doesn't do anything if the device wasn't created with the MULTITHREAD flag.